### PR TITLE
Event search based on dates

### PR DESCRIPTION
**Pull request by k-n185**
DIdn't create eventBTO because there are only 2 fields that we need to ignore from EventDTO. I will manage the fields in binding with thymeleaf

### CODE REVIEW

File: src/main/java/edu/txstate/cs/repository/EventRepo.java [hunk 1]
```diff
@@ -4,6 +4,11 @@
 
 import edu.txstate.cs.model.dto.Event;
 
+import java.time.LocalDate;
+import java.util.List;
+
 public interface EventRepo extends JpaRepository<Event, Long>{
 
+    public List<Event> findByDateBetween(LocalDate after, LocalDate before);
```
Comment (X625) [inline src/main/java/edu/txstate/cs/repository/EventRepo.java:12]: r u sure it is after, before? and not before, after? have u tested?

File: src/main/java/edu/txstate/cs/service/EventService.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import java.time.LocalDate;
+import java.util.List;
+
+@Service
+public class EventService {
+
+    @Autowired
+    private EventRepo repo;
+
+    public List<Event> getEventsBeforeAndAfter(LocalDate before, LocalDate after)
```
Comment (X625) [inline src/main/java/edu/txstate/cs/service/EventService.java:17]: It is ok for this project to send back DTO object, but don't do it in a real project. it exposes security issues.

File: src/main/java/edu/txstate/cs/repository/EventRepo.java (hunk 1 above)
Comment (k-n185) [reply src/main/java/edu/txstate/cs/repository/EventRepo.java:12]: Yeah it goes like - if we need events between Jan 2020 and December 2020, we have to search **after** Jan 2020 and **before** December 2020

between works in that way.

File: src/main/java/edu/txstate/cs/service/EventService.java (hunk 2 above)
Comment (k-n185) [reply src/main/java/edu/txstate/cs/service/EventService.java:17]: Got it :)

**Comment by k-n185** [comment]
Tested OK
image

**Pull request merged**